File: Programmers/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i=0;i<commands.size();i++){
        int n = commands[i][0] - 1;
        int m = commands[i][1] - 1;
        vector<int> temp;
        for(int j=n;j<=m;j++){
            temp.push_back(array[j]);
        }
        sort(temp.begin(), temp.end());
        answer.push_back(temp[commands[i][2]-1]);
    }
    
    return answer;
}

File: Programmers/K번째수.java
import java.util.*;
public class K번째수 {
    public int[] solution(int[] array, int[][] commands) {
        int n = commands.length;
        int[] answer = new int[n];
        for(int i=0;i<n;i++) {
            int x = commands[i][0]-1;
            int y = commands[i][1];
            int[] temp = Arrays.copyOfRange(array,x,y);
            Arrays.sort(temp);
            answer[i] = temp[commands[i][2]-1];
        }
        return answer;
    }
}
